### PR TITLE
Prompt compiler

### PR DESCRIPTION
**Pull request by antonieto**

### CODE REVIEW

File: server/adapters/prompt_compiler/compiler.y [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+%start Prompt
+
+%%
+Prompt:
+	  InstructionPrompt {
+		// Print as stringified JSON in one line
+		printf("{\"verb\": \"%s\", \"description\": \"%s\", \"payload\": \"%s\"}\n", $1.verb, $1.description, $1.payload);
+
+	  }
```
Comment (CarlosCRG19) [inline server/adapters/prompt_compiler/compiler.y:43]: is this going to be the same body for an `assertion`?

File: server/adapters/prompt_compiler/lexer.lex [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+%%
+{ClickLiteral}|{TapLiteral}		{
+	yylval.str = strdup(yytext);
+	return CLICK_VERB;
+}
+
+{TypeLiteral}|{WriteLiteral}	{
+	yylval.str = strdup(yytext);
+	return TYPE_VERB;
+}
```
Comment (CarlosCRG19) [inline server/adapters/prompt_compiler/lexer.lex:31]: good soup

File: server/Makefile [hunk 3]
```diff
@@ -27,3 +27,9 @@ stop-docker:
 # Remove Docker image
 clean-docker:
 	docker rmi $(DOCKER_IMAGE_NAME)
+
+# Lex Yacc build
+build-compiler:
+	lex  adapters/prompt_compiler/lexer.lex
+	yacc adapters/prompt_compiler/compiler.y
+	gcc -o adapters/prompt_compiler/compiler adapters/prompt_compiler/y.tab.c adapters/prompt_compiler/lex.yy.c -ll
```
Comment (CarlosCRG19) [inline server/Makefile:35]: need to specify the output file for lex and yacc. also `-ll` option might be changed to `-lfl` depending on the lex implementation you are using (victor mentioned it in class, i actually had to change it to `-lfl`). This makefile was generated by ChatGPT and it checks conditionally if you have to use `-ll` or `-lfl` dunno exactly what it is doing but it works so you might want add it or not (the only mandatory change would be the specification of output files)
```suggestion
build-compiler: check-lib
	lex -o adapters/prompt_compiler/lex.yy.c adapters/prompt_compiler/lexer.lex
	yacc -o adapters/prompt_compiler/y.tab.c adapters/prompt_compiler/compiler.y
	gcc -o adapters/prompt_compiler/compiler adapters/prompt_compiler/lex.yy.c adapters/prompt_compiler/y.tab.c $(LEX_LIB)

check-lib:
	@echo "Checking for lex library..."
	@if ldconfig -p | grep -q libfl; then \
		echo "Using -lfl"; \
		LEX_LIB=-lfl; \
	elif ldconfig -p | grep -q libl; then \
		echo "Using -ll"; \
		LEX_LIB=-ll; \
	else \
		echo "Error: Neither libfl nor libl found"; \
		exit 1; \
	fi
	@echo "LEX_LIB=$(LEX_LIB)" > lex_lib.mk

-include lex_lib.mk
```

**Review by CarlosCRG19 — APPROVED** [review]
LGTM! so right now we are not supporting assertions are we? is there a way in which we can handle assertions as just `instructions` for the first version?

maybe we can ignore the `ASSERTION_END` and leave it as part of the description (e.g. "light red circle containing a heart emoji is on the page")... it might be risky and it might obscure the results of the SS model, but its the first option that came to mind in order to rapidly include `assert`/`verify` as an option

File: server/adapters/prompt_compiler/compiler.y (hunk 1 above)
Comment (antonieto) [reply server/adapters/prompt_compiler/compiler.y]: i discussed this with @BrandonMagana and we thought returning `type  | click | assert` could work, what do you think?

**Review by Int-Angel — APPROVED** [review]
great work

**Review by BrandonMagana — APPROVED** [review]
Nice job.
 
Why do you keep the `y.tab.h` file? Isn't it generated while building the compiler?

**Pull request merged**